source/server-selection/server-selection.rst

```diff
@@ -1705,17 +1732,42 @@ The user's intent in specifying two tag sets was to fall back to the second set
 if needed, so we filter by maxStalenessSeconds first, then tag_sets, and select
 Node 2.
 
+Why change from pure random selection when selecting from within the latency window?
+------------------------------------------------------------------------------------
+
+When a node slows down, pooled connections to it will remain checked out for
+longer periods of time due to operations taking longer to complete
+server-side. Assuming at least constant incoming operation load, more
+connections will need to be opened against the node to service new operations,
+further straining it and slowing it down. This can lead to runaway connection
+creation scenarios that can cripple a deployment ("connnection storms"). As part
+of DRIVERS-781, the load balancing algorithm was changed to more evenly spread
+out the workload among suitable servers to prevent any single node from being
+overloaded. The new algorithm achieves this by routing operations to servers
+with fewer active connections, thereby reducing the number of new operations and
+connection creations routed towards nodes that are busier. The previous random
+selection mechanism did not take load into account and could assign work to
+nodes that were under too much stress already.
+
+As an added benefit, the new algorithm gives preference to nodes that have
+recently been discovered and are thus are more likely to be alive (e.g. during a
+rolling restart). The narrowing to two random choices first ensures new servers
+aren't overly preferred however, preventing a "thundering herd" situation.
+
 References
 ==========
 
 - `Server Discovery and Monitoring`_ specification
 - `Driver Authentication`_ specification
+- `Connection Monitoring and Pooling`_ specificaiton
 
 .. _Server Discovery and Monitoring: https://github.com/mongodb/specifications/tree/master/source/server-discovery-and-monitoring
 .. _heartbeatFrequencyMS: https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#heartbeatfrequencyms
 .. _Max Staleness: https://github.com/mongodb/specifications/tree/master/source/max-staleness
 .. _idleWritePeriodMS: https://github.com/mongodb/specifications/blob/master/source/max-staleness/max-staleness.rst#idlewriteperiodms
 .. _Driver Authentication: https://github.com/mongodb/specifications/blob/master/source/auth
+.. _Connection Pool: https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#connection-pool
```

Comment:
Can you use a relative links here? The benefit is that the links will work correctly for PRs, branches, and forks. Like this:
```
.. _Connection Pool: /source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#connection-pool
.. _Connection Monitoring and Pooling: /source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#connection-pool
```

Reply:
done